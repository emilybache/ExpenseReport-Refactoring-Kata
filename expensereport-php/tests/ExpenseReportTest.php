<?php

namespace Expenses\Test;

use Expenses\Expense;
use Expenses\ExpenseReport;
use Expenses\ExpenseType;
use PHPUnit\Framework\TestCase;
use ApprovalTests\Approvals;

class ExpenseReportTest extends TestCase {

    function createExpense($type, $amount): Expense
    { return new Expense($type, $amount); }

    public function testReport()
    {
        ob_start();

        $expenses = [
            $this->createExpense(ExpenseType::DINNER, 5000),
            $this->createExpense(ExpenseType::DINNER, 5001),
            $this->createExpense(ExpenseType::BREAKFAST, 1000),
            $this->createExpense(ExpenseType::BREAKFAST, 1001),
            $this->createExpense(ExpenseType::CAR_RENTAL, 200)
        ];
        $report = new ExpenseReport();
        $report->print_report($expenses);

        $output = ob_get_clean();
        // drop the first line because it varies. Crude but effective.
        $output = preg_replace('/^.+\n/', '', $output);
        Approvals::verifyString($output);
    }
}


