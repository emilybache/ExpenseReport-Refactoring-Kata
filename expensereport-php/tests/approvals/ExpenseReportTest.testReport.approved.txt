Dinner	5000	 
Dinner	5001	X
Breakfast	1000	 
Breakfast	1001	X
Car Rental	200	 
Meal Expenses: 12002
Total Expenses: 12202
